langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -92,77 +208,247 @@ public void removeAll(Collection<String> ids) {
+                            .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+
+                    preparedStatement.setString(1, id);
+                    preparedStatement.setObject(2, embedding);
+                    preparedStatement.setString(3, text);
+                    int j = 4;
+                    if (embeddedMetadataCopy != null && !embeddedMetadataCopy.isEmpty()) {
+                        for (; j < metadataColumns.size(); j++) {
+                            if (embeddedMetadataCopy.containsKey(metadataColumns.get(j).getName())) {
+                                preparedStatement.setObject(j, embeddedMetadataCopy.remove(metadataColumns.get(j).getName()));
+                            } else {
+                                preparedStatement.setObject(j, null);
+                            }
+                        }
+                        if (isNotNullOrEmpty(metadataJsonColumn)) {
+                            // metadataJsonColumn should be the last column left
+                            preparedStatement.setObject(j, OBJECT_MAPPER.writeValueAsString(embeddedMetadataCopy), Types.OTHER);
+                        }
+                    } else {
+                        for (; j < metadataColumns.size(); j++) {
+                            preparedStatement.setObject(j, null);
+                        }
+                    }
+                    preparedStatement.addBatch();
+
+                }
+                preparedStatement.executeUpdate();
+
+            } catch (JsonProcessingException ex) {
+                log.error("Exception caught when inserting into vector store table: \"" + schemaName + "\".\"" + tableName + "\"", ex);
+                throw new RuntimeException(ex);
+            }
+
+        } catch (SQLException ex) {
+            log.error("Exception caught when inserting into vector store table: \"" + schemaName + "\".\"" + tableName + "\"", ex);
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
     public static class Builder {
 
-        private AlloyDBEngine engine;
         private String tableName;
         private String schemaName;
         private String contentColumn;
         private String embeddingColumn;
         private String idColumn;
-        private List<String> metadataColumns;
+        private List<MetadataColumn> metadataColumns;
         private String metadataJsonColumn;
-        private List<String> ignoreMetadataColumns;
+        private List<String> ignoreMetadataColumnNames;
+        private DistanceStrategy distanceStrategy;
+        private Integer k;
+        private Integer fetchK;
+        private Double lambdaMult;
         // change to QueryOptions class when implemented
         private List<String> queryOptions;
+        private AlloyDBEngine engine;
 
         public Builder() {
```

Comment:
We want table name and engine to be required:
```suggestion
        public Builder(AlloyDBEngine engine, String tableName) {
```
Then remove the set methods. The default values should be set in the list of attributes not in the builder

Reply:
done